youtube_dl/extractor/dumpert.py

```diff
@@ -39,18 +59,24 @@ def _real_extract(self, url):
         title = item['title']
         media = next(m for m in item['media'] if m.get('mediatype') == 'VIDEO')
 
-        quality = qualities(['flv', 'mobile', 'tablet', '720p'])
+        quality = qualities(['flv', 'mobile', 'tablet', '720p', '1080p'])
         formats = []
         for variant in media.get('variants', []):
             uri = variant.get('uri')
             if not uri:
                 continue
             version = variant.get('version')
-            formats.append({
-                'url': uri,
-                'format_id': version,
-                'quality': quality(version),
-            })
+            ext = determine_ext(uri)
+            if ext == 'm3u8':
+                formats.extend(self._extract_m3u8_formats(
+                    uri, video_id, ext='mp4', m3u8_id=version))
+            else:
+                formats.append({
+                    'url': uri,
+                    'format_id': version,
+                })
+            for format in formats:
+                format['quality'] = quality(format['format_id'])
```

Comment:
So, looking at (1 of) the manifests, they're playlists and you could pass `preference=quality(version)` into `_extract_m3u8_formats()`. Probably also `entry_protocol='m3u8_native' as the manifests look harmless.

Incidentally the API data appears to be in the webpage hydration JSON as well `window.__DUMPERT_STATE__ = JSON.parse(...)` but as you don't look at the page itself that doesn't matter.

Reply:
Hm, for some weird reason I thought I'd seen somewhere that preference was supposed to be a float between 0.0 and 1.0 and that seemed annoying/ugly with a possibly longer list of qualities. Obviously it's free-form. I'll change it to preference in both cases, and add the suggested entry_protocol change.

Reply:
For some reason, using entry_protocol='m3u8_native' changes the md5. Calling youtube-dl with --test uses ffmpeg to download a file of approx 93kB which is then used for the md5, rather than only the first 10241 bytes as claimed in the docs. After some debugging in the tests code it seems that it does use a 10241-bytes long file to calculate the md5 from, however it seems to have some kind of header prepended that the resulting video does not have. I'm just going to assume everything is all right as the resulting complete files obtained through both methods are identical.

Reply:
Generally, downloads using a manifest might be adaptive so the file MD5 check may not be reliable. Following precedent I put `'params': {'skip_download': True},` in the test case and comment out the `'skip_download'` line while testing locally.

Reply:
In this case, keep it as is then?